app/_src/gateway/reference/wasm.md

```diff
@@ -0,0 +1,147 @@
+typically spans several different Lua source code files, all packaged and
+deployed in text format. Filters must be compiled from source code
+into a single binary file with the `.wasm` extension.
+
+In order to be usable in {{site.base_gateway}}, all filter files must be present
+on disk in a single directory, denoted by the `wasm_filters_path` configuration
+directive.
+
+### Filter Chain
+
+A Filter Chain is the database entity representing one or more filters executed
+for each request to a particular Service or Route, each one with its
+configuration.
+
+Example:
+
+```yaml
+---
+filter-chains:
+  name: my-filter-chain
+  # Filter Chains can be toggled on/off for testing
+  enabled: true
+  filters:
+    # this maps to $wasm_filters_path/my_filter.wasm
+    - name: my_filter
+      # at runtime, configuration is passed to the filter as a byte array
+      #
+      # filter configuration is inherently untyped--schema and
+      # encoding/serialization are left up to the filter code itself
+      config: "foo=bar baz=bat"
+
+    - name: my_other_filter
+      # individual filters within a chain can be toggled on/off for testing
+      enabled: false
+      config: >-
+          {
+            "some_settings": {
+              "a": "b",
+              "c": "d"
+            },
+            "foo": "bar"
+          }
+
+    # the same filter may occur more than once in a single chain, and all   entries will be
+    # executed
+    - name: my_filter
+      config: "foo=123 baz=456"
+```
+
+
+#### Relationships
+
+Filter Chains **must** be linked to a Service or Route, and as of this writing,
+the relationship is one-to-one:
+
+* a Service or Route may only be linked to a single Filter Chain
+* a Filter Chain may only be linked to a single Service or Route
+
+## Filter execution behavior
+
+The filter execution plan for a given request is determined during the `access`
+phase, after Lua plugins execution. If the Service entity for the request
+is linked to a Filter Chain, the filters are executed. If the Route entity that
+was matched to the request is linked to a Filter Chain, its filters are also
+executed, _after_ Service-related filters (if any) have been executed. You can
+inspect the  full  filter execution plan for a given Route via the Admin API,
+through the endpoint `/routes/{route name or id}/filters/enabled`.
+
+Filters are always executed in the order in which they are defined in the filter
+chain. The same filter may appear in the execution plan multiple times, either
+coming from the Service and Route chains, or even within the same Filter Chain.
+Each entry will execute with its own configuration.
+
+### Can Lua plugins and Proxy-Wasm filters be used at the same time?
+
+Yes. Keep in mind though that for request phase, Lua plugins are executed
+_before_ Wasm filters.
+
+There is one crucial side effect of this ordering to be aware of: Filter Chain
+attachment is assessed during the `access` phase, _after_ Lua plugin execution.
```

Comment:
"Filter Chain attachment" would read better if it were a clear term already used in the above description of Filter Chains behavior. Otherwise without prior context, it reads like an implementation detail at best, or some incompleteness at worst.

A couple thoughts that come to mind is to maybe rename one of the above section to something mentioning "attach", maybe: "How are Filter Chains attached to a request?" or "Filter Chains attachment behavior", and have another section be "Filter Chains execution behavior".

And/or a clear mention of "attaching" Filter Chains to a request for "later execution after all plugins handlers". Or even maybe a couple bold words reading "this Filter Chain is now considered **attached** to the request". 

None of these are specific suggestions but only trying to carry my point, does it make sense?

Reply:
*Disable this warning with `-Wno-used-but-not-formalized`*